### ISSUE TITLE
SameDiff logsoftmax evaluation

### ISSUE THREAD

**Issue by lastrealklaus** [deeplearning4j/deeplearning4j#7849]
i would expect that calculating logsoftmax of same input results in same output. but when i run this:

	public static void main(String[] args) {
		final var inputArr = Nd4j.rand(new int[] { 1, 4 }, new NormalDistribution());
		System.err.println();
		System.err.println("created input array: " + inputArr);
		System.err.println("nd4j.oldlogsoftmax:  " + Nd4j.getExecutioner().exec(new OldLogSoftMax(inputArr.dup())));

		for (int i = 0; i < 10; i++) {
			SameDiff sd = SameDiff.create();
			final var input = sd.var(inputArr.dup());
			var softmax = sd.nn().logSoftmax(input);
			System.err.println("samediff.logsoftmax: " + softmax.eval());
		}
	}

the output may look like this:

```
created input array: [[    0.1869,   -1.4918,   -0.6497,   -0.8864]]
nd4j.oldlogsoftmax:  [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[       NaN,       NaN,       NaN,       NaN]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[       NaN,       NaN,       NaN,       NaN]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]
samediff.logsoftmax: [[       NaN,       NaN,       NaN,       NaN]]
samediff.logsoftmax: [[   -0.6738,   -2.3525,   -1.5104,   -1.7472]]

```
the output varies as do numbers of NaN outputs.

tried it with 1.0.0-beta4 and snapshot build of today.

**Comment by raver119** [comment]
Which backend was used to produce this? CPU or CUDA?

**Comment by lastrealklaus** [comment]
i'm using native backend. can i use the cuda backend for samediff? thought i read, that this is currently not available.

**Comment by raver119** [comment]
Please don't. I'm just trying to imagine what could go wrong, and backend used is the first question I should've asked.

**Comment by lastrealklaus** [comment]
while looking for a workaround i found another problem:

		final var inputArr = Nd4j.rand(new int[] { 1, 4 }, new NormalDistribution());
		final var sd = SameDiff.create();
		final var input3 = sd.var(inputArr.dup());
		final var custom2 = sd.math().logSumExp(input3);
		System.err.println("samediff.logSumExp:  " + custom2.eval());

prints

```
samediff.logSumExp:  0
[ERROR] Unknown opNum=11 on D:/jenkins/ws/dl4j-master-windows-x86_64-cpu/libnd4j/include/loops/cpu/reduce/reduce_float.cpp:124
```

**Comment by raver119** [comment]
Hm. That's separate issue. Just a bad mapping. Open new issue about this one please.

**Comment by lastrealklaus** [comment]
done

**Comment by raver119** [comment]
Thank you. This issue is already fixed, but fix will be merged only after migration process finished.

**Comment by shyrma** [comment]
Hi, guys 

thanks for catching  this
logSoftMax works correctly now
